Create cards select

JosephPotashnik opened the pull request.

RedHellier reviewed (approved):
Very nice, going to be some cheeky conflicts with mine and Anna's stuff from yesterday but I'll ask you if I have any questions about that.

JosephPotashnik commented:
Thank you! @RedHellier 

Hopefully the conflicts will be very limited. 
In terms of topic.ts file, 

1) I took out the supabase client to an external file because it's used in the card model too.
 import { createSupabaseClient } from "../supabase/client";
 
 2) I commented out - same as you did, my getTopic() experimental logic that called unnecessary stuff, returned only json(allTopics). you can absolutely take your own version of getTopics() here and override mine.
 
 3) I added export in front of every function. should be trivial.
 
4) other than that I think you worked on new functions which I don't have so that should be easy and distinct. 
 I added new functions getTopicByTitle, getSubtopicByTopicIdAndSubtopicTitle which are necessary for the cards.
